truss/templates/shared/logging.py

```diff
@@ -1,63 +1,110 @@
+import json
 import logging
 import sys
+from datetime import datetime
+from enum import Enum
 
+import loguru
 from pythonjsonlogger import jsonlogger
 
 LEVEL: int = logging.INFO
 
-JSON_LOG_HANDLER = logging.StreamHandler(stream=sys.stdout)
-JSON_LOG_HANDLER.set_name("json_logger_handler")
-JSON_LOG_HANDLER.setLevel(LEVEL)
-JSON_LOG_HANDLER.setFormatter(
-    jsonlogger.JsonFormatter("%(asctime)s %(levelname)s %(message)s")
-)
+
+class Lifecycle(Enum):
+    LOAD = "MODEL_LOAD"
+    STARTUP = "MODEL_STARTUP"
+    REQUEST = "REQUEST"
+
+
+def serialize(record):
+    dt = datetime.fromtimestamp(record["time"].timestamp())
+    formatted_time = dt.strftime("%Y-%m-%d %H:%M:%S,%f")[:-3]
+
+    request_id = (
+        str(record["extra"]["request_id"])
+        if "request_id" in record["extra"] and record["extra"]["request_id"] is not None
+        else None
+    )
+    lifecycle = (
+        record["extra"]["lifecycle"].value
+        if "lifecycle" in record["extra"] and record["extra"]["lifecycle"] is not None
+        else None
+    )
+
+    # if someone is using logger instead of print, we need to handle that
+    try:
+        json_message = json.loads(record["message"])
```

Comment:
it sounds like from this logic if a normal logger is being used, record["message"] will be json, and we should parse that json and extract the `message` field.

Could we write this to make that clearer? Maybe have an `is_json_object` function that checks if it's json (that could do the `try/except`, and then use that function here to determine how to set `record["message"]`

Reply:
Added!